### ISSUE TITLE
Add squish to the Transform Text dialogue

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#6705]
This is a small task.  It isn't urgent, but might be a useful improvement for an intern given they have been improving the text keyboard in the calculator.

![image](https://user-images.githubusercontent.com/11226469/130780903-3569e625-2635-4064-b651-295e89437ba4.png)
This is the current dialogue for trimming spaces.
There are 3 buttons for Left, Right and Both.
There could usefully be a 4th option Maybe call it Within (squish).  It would execute the squish command, rather than trim.  

That's it!

At the same time please check that the code is generally ok.  This was all written some time ago.  I assume it uses stringr sensibly?  I think so.

**Comment by Gioche6** [comment]
@shadrackkibet , Yes, I will take up the challenge. Thank you, I will constantly seek guidance from you and the team whenever I come across a challenge beyond my reach.

**Comment by Ivanluv** [comment]
@shadrackkibet  As I have been helping out with this I realized that the code has some implementation inconsistencies eg having the `setRcodeForControls `run within `ucrPnl_ControlValueChanged `sub. Should I 
refactor the code to our normal implemetation then they can go on with this issue?

**Comment by shadrackkibet** [comment]
@Ivanluv that will be nice to sort out. If it doesn't take much time please go ahead with refactoring. Otherwise adding the Squish feature is the priority for now.

**Comment by Ivanluv** [comment]
@shadrackkibet this wont take  much time

**Comment by shadrackkibet** [comment]
Go for it!

**Comment by rdstern** [comment]
@Ivanluv and @shadrackkibet just to support this improving-of-the-code step, that you plan, or perhaps have done already.  It is important and also good to add for training.  We need to make general improvements to the prepare menu, and having consistent, high-quality code is an important component.
Here this also links with making good use of the tidyr family of packages - this time stringr.  We can now do most of stringr using the calculator.  These dialogues are partly to make it easier to do those tasks, and also to add facilities that are not possible with the calculator.  (It is limited to transformations that produce another variable of the same "length".
